log updated records

fenekku opened the pull request:
- [#641] add logging for updated records

fenekku commented on galter_subjects_utils/writer.py:
```diff
@@ -26,3 +29,89 @@ def write_jsonl(
+        """Constructor."""
+        if not filepath:
+            # In memory
+            self.f = StringIO()
```
Really a convenience for tests. It has the same interface as File objects which allows us to do nice things.

fenekku commented on galter_subjects_utils/writer.py:
```diff
@@ -26,3 +29,89 @@ def write_jsonl(
+
+        if delta:
+            if delta["type"] == "replace":
+                delta_msg = f"{delta['id']} -> {delta['new_id']}"
+            elif delta["type"] == "remove":
+                delta_msg = f"{delta['id']} -> X"
```
Admittedly "cute" representation of changes for human consumption. Meh :shrug: .

fenekku commented on galter_subjects_utils/writer.py:
```diff
@@ -26,3 +29,89 @@ def write_jsonl(
+        offset = self.f.tell()
+        self.f.seek(0)
+        result = [e for e in csv.DictReader(self.f)]
+        self.f.seek(offset)
```
The aforementioned nice things.

fenekku commented on tests/test_updater.py:
```diff
@@ -252,4 +261,68 @@ def test_update_on_records(update_result, records_data_w_subjects):
+    try:
+        raise Exception("msg")
+    except Exception as e:
+        logger.log("abcde-54321", delta=delta, error=str(e))
```
I will actually change this to be 

```
logger.log("abcde-54321", error=str(e))
```

because that's really how it is used.

mattbaumann1 reviewed (approved):
Looks good. Thank you for working on the logger.

Meowcenary reviewed (approved):
I definitely understood this one 😄 I think Django has logging utilities built in, but I might reference this PR when I loop back around to that at some point.

fenekku commented:
I've moved the purely logger focused test in test_writer and changed the `log` call in that test to only pass an error as commented before. Will merge this soon.
